extension/js/newResponseNotification.js

```diff
@@ -1,63 +1,85 @@
+  if (!backgroundSync || !backgroundSyncPosts) return;
 
-                    let markUnreadUrls = (data.match(/\/forum\/app\.php\/markpostunread\/\d+\/\d+/g) || []);
-                    if(markUnreadUrls.length > currentCount)
-                        fetch(markUnreadUrls[currentCount]);
-                    
-                        setInterval(function(){                                   
-                            document.title = (document.title == title ? '\u26B9 ' + title : title);
-                        }, 500);
-                }else if(isNewPage){
-                    lastPost.insertAdjacentHTML('afterend', 
-                        `<h3 style="margin:4px auto;text-align:center;background:#efeed9;padding:5px;border:none;border-radius:7px;user-select:none;">
-                            א נייע בלאט איז צוגעקומען
-                            <a class="button" style="width:150px;margin:6px auto 5px auto;display:block;" ${(data.match(/href=".*" rel="next"/g) || [])[0]}>גיי צום קומענדיגן בלאט</a>
-                        </h3>`)
-                        setInterval(function(){                                   
-                            document.title = (document.title == title ? '\u26B9 ' + title : title);
-                        }, 500);
-                }
-                else{
-                    setTimeout(checkNewResponse, interval);
-                }
-            }).catch(function (error) {
-                    interval *= 3;
-                    setTimeout(checkNewResponse, interval);
-                }
-            )
-        }
+  let interval = parseInt(backgroundSyncPosts);
+  let checkNewResponse = function () {
+    if (isLastPage() && currentCount > 0) {
+      fetch(forumURL)
+        .then(function (response) {
+          return response.text();
+        })
+        .then(function (data) {
+          var parser = new DOMParser();
+          var doc = parser.parseFromString(data, "text/html");
+          let topics = Array.from(
+            Array.from(doc.getElementsByClassName("forumbg"))
+              .pop()
+              .getElementsByClassName("topics")[0]
+              .querySelectorAll("li.row")
+          );
+          let topic = topics.find((t) => {
+            let tHref = t.children[0]?.children[0].children[0].href;
+            if (!tHref) return false;
+            let tHrefSearch = tHref.substr(tHref.indexOf("?"));
+            let tSearch = new URLSearchParams(tHrefSearch);
+            let tId = tSearch.get("t");
+            if (tId == topicId) return true;
+            else return false;
+          });
+          //If we cannot find the topic on the first page of the forum we can safely assume that no new post exists.
+          if (!topic) return setTimeout(checkNewResponse, interval);
+          let newCount =
+            topic.children[0]?.children[1]?.firstChild?.textContent?.trim();
+          if (currentCount < newCount * 1 + 1) {
+            lastPost.insertAdjacentHTML(
```

Comment:
it would be maybe cool to load the new content with ajax and add to the page so its there for the user ready to read... maybe its too advanced...

Reply:
Then again you run into the unread-tracking interference. Unless you only load it when the user click on the "show" button but then you've lost most of the benefits of preparing it beforehand. In short, I do not recommend.